src/__tests__/rosie.test.js

```diff
@@ -635,6 +701,54 @@ describe('Factory', () => {
         );
         expect(useCapsLockValues).toEqual([false, true]);
       });
+
+      it('can depend on other options', () => {
+        factory
+          .option('option1', 'foo')
+          .option('option2', ['option1'], (option1) => option1 + 'bar')
+          .attr('value', ['option2'], (option2) => option2);
+
+        // Default values
+        expect(factory.attributes()).toHaveProperty('value', 'foobar');
+        // Override one
+        expect(factory.attributes({}, { option1: 'bar' })).toHaveProperty(
+          'value',
+          'barbar'
+        );
+        // Override two
+        expect(factory.attributes({}, { option2: 'specific' })).toHaveProperty(
+          'value',
+          'specific'
+        );
+      });
+
+      it('cannot depend on itself', () => {
+        const fn = jest.fn().mockReturnValue('default value');
+        factory
+          .option('option', ['option'], fn)
+          .attr('value', ['option'], (option) => option);
+
+        // Default values
+        expect(() => factory.attributes()).toThrow(/Maximum call stack/);
+        expect(fn).not.toHaveBeenCalled();
+        // Option set
+        expect(factory.attributes({}, { option: 'override' })).toHaveProperty(
+          'value',
+          'override'
+        );
+        expect(fn).not.toHaveBeenCalled();
+      });
+
+      it('cannot depend on an attribute', () => {
+        factory
+          .attr('baseAttr', 'base')
+          .option('option', ['baseAttr'], (attr) => attr)
+          .attr('value', ['option'], (option) => option);
+
+        expect(() => factory.attributes()).toThrow(
+          /Cannot read properties of undefined/
```

Comment:
Looks like Node v14 has an error that doesn't match this regex. Use this instead?
```suggestion
          /Cannot read (properties|property).* of undefined/
```
It probably doesn't make sense to continue supporting Node 14 since it was EOL'd in April, but that's not something I want to do as part of this PR.

Reply:
Fixed. Would you be interested in a second PR that throws a more useful error in these cases?